lib/document/validateDocumentFactory.js

```diff
@@ -106,6 +110,25 @@ module.exports = function validateDocumentFactory(
       );
     }
 
+    if (!result.isValid()) {
+      return result;
+    }
+
+    const documentId = bs58.encode(
```

Comment:
Looks like a code duplication (the same in factory). Let’s extract it to a separate function?